pages/api/applications/index.tsx

```diff
@@ -206,6 +206,73 @@ async function handlePostApplications(req: NextApiRequest, res: NextApiResponse)
   });
 }
 
+/**
+ * Handles PUT requests to /api/applications.
+ *
+ * This updates an existing application in the database using the information
+ * provided in the request. If a user is not signed in, this route will return a
+ * 403 Unauthenticated error.
+ *
+ * @param req The HTTP request
+ * @param res The HTTP response
+ */
+async function handlePutApplications(req: NextApiRequest, res: NextApiResponse) {
+  const registrationAllowed = await checkRegistrationAllowed();
+  if (!registrationAllowed) {
+    return res.status(403).json({
+      msg: 'Registration updates are no longer allowed',
+    });
+  }
+
+  const applicationBody = req.body;
+
+  let body: Registration;
+  try {
+    body = JSON.parse(req.body);
+  } catch (error) {
+    console.error('Could not parse request JSON body');
+    return res.status(400).json({
+      type: 'invalid',
+      message: 'Invalid JSON body',
+    });
+  }
+
+  const snapshot = await db
+    .collection(APPLICATIONS_COLLECTION)
+    .where('user.id', '==', body.user.id)
+    .get();
+
+  if (snapshot.empty) {
+    return res.status(404).json({
+      msg: 'Profile does not exist',
+    });
+  }
+
+  const completedRegistrationInfo = {
+    ...body,
+    user: {
+      ...body.user,
+      permissions: ['hacker'],
```

Comment:
Might need to do something else here so that we won't accidentally override organizer's permissions.